lib/rdoc/parser/prism_ruby.rb

```diff
@@ -510,13 +510,21 @@ def add_method(name, receiver_name:, receiver_fallback_type:, visibility:, singl
 
     receiver = receiver_name ? find_or_create_module_path(receiver_name, receiver_fallback_type) : @container
     meth = RDoc::AnyMethod.new(nil, name, singleton: singleton)
+
     if (comment = consecutive_comment(start_line))
       handle_consecutive_comment_directive(@container, comment)
       handle_consecutive_comment_directive(meth, comment)
 
       comment.normalize
       meth.call_seq = comment.extract_call_seq
+
+      # Save visibility before comment assignment (where YARD processing happens)
+      visibility_before = meth.visibility
       meth.comment = comment
+      # Check if YARD tags changed the visibility
+      if meth.visibility != visibility_before
+        visibility = meth.visibility
+      end
```

Comment:
It seems that this shows that the proposed approach isn't integrated with the current implementation clearly. Can we use another more integrated approach?